test/Interop/Cxx/value-witness-table/custom-virtual-destructors.swift

```diff
@@ -0,0 +1,61 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop -Xcc -fno-rtti)
+//
+// REQUIRES: executable_test
+// Windows doesn't support -fno-rtti.
+// UNSUPPORTED: OS=windows-msvc
+
+import CustomDestructorTypes
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXXDestructor")
+
+@inline(never)
+func createTypeWithVirtualBaseAndDestructor(
+  _ ptr: UnsafeMutablePointer<Int32>
+) {
+  _ = HasVirtualBaseAndDestructor(ptr)
+}
+
+@inline(never)
+func createTypeWithBaseWithVirtualDestructor(
+  _ ptr: UnsafeMutablePointer<Int32>
+) {
+  _ = HasBaseWithVirtualDestructor(ptr)
+}
+
+@inline(never)
+func createTypeWithVirtualBaseWithVirtualDestructor(
+  _ ptr: UnsafeMutablePointer<Int32>
+) {
+  _ = HasVirtualBaseWithVirtualDestructor(ptr)
+}
+
+CXXDestructorTestSuite.test("Virtual base and destructor") {
+  let ptr = UnsafeMutablePointer<Int32>.allocate(capacity: 1)
+  ptr.pointee = 0
+  createTypeWithVirtualBaseAndDestructor(ptr)
+  expectEqual(ptr.pointee, 42)
+  ptr.deallocate()
```

Comment:
```suggestion
  var value: Int32 = 0
  createTypeWithVirtualBaseAndDestructor(&value)
  expectEqual(value, 42)
```

Similarly in other tests where the pointer does not escape a single call.